subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CacheTaskArchiveErrorIntegrationTest.groovy

```diff
@@ -380,6 +381,32 @@ class CacheTaskArchiveErrorIntegrationTest extends AbstractIntegrationSpec {
         api = useRuntimeApi ? "runtime" : "annotation"
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/9906")
+    def "don't cache if task produces broken symlink"() {
+        buildFile << """
+            import java.nio.file.*
+            class ProducesLink extends DefaultTask {
+                @OutputDirectory File outputDirectory
+
+                @TaskAction execute() {
+                    Files.createSymbolicLink(Paths.get('root/link'), Paths.get('target'));
+                }
+            }
+
+            task producesLink(type: ProducesLink) {
+                outputDirectory = file 'root'
+                outputs.cacheIf { true }
+            }
+        """
+
+        when:
+        executer.withStackTraceChecksDisabled().withBuildCacheEnabled()
+        run "producesLink"
+        then:
+        executedAndNotSkipped ":producesLink"
+        outputContains "Couldn't read file content 'tree-outputDirectory/link'"
```

Comment:
The error message should contain the absolute path of the file we couldn't read and not the relative path it will have in the archive.

Reply:
Done, thanks